device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubSendTask.java

```diff
@@ -11,30 +13,43 @@
 {
     protected final IotHubTransport transport;
 
+    /**
+     * Private logger for class
+     */
+    private final CustomLogger logger = new CustomLogger(this.getClass());
+    
     public IotHubSendTask(IotHubTransport transport)
     {
         // Codes_SRS_IOTHUBSENDTASK_11_001: [The constructor shall save the transport.]
         this.transport = transport;
+        
+        if (transport == null)
+        	throw new IllegalArgumentException("Parameter 'transport' must not be null");
```

Comment:
@prmathur-microsoft : With current setup constructors of IotHubSendTask and IotHubReceiveTask will write a wrong log error message when parameter 'transport' is not null. If it's null, no log message will be written, since the code is no longer reachable